sys/linux/socket_netlink_generic_devlink.txt

```diff
@@ -71,3 +71,61 @@ devlink_port_set {
 	port_handle		devlink_port_handle
 	DEVLINK_ATTR_PORT_TYPE	nlattr[DEVLINK_ATTR_PORT_TYPE, int16[DEVLINK_PORT_TYPE_NOTSET:DEVLINK_PORT_TYPE_IB]]
 } [packed, align_4]
+
+type msghdr_nl_devlink_trap_set msghdr_netlink[netlink_msg_t[genl_devlink_family_id, genlmsghdr_t[DEVLINK_CMD_TRAP_SET], devlink_trap_set]]
```

Comment:
I see this file does the same above, but this leads to some duplication.
A better way to do this is:
https://github.com/google/syzkaller/blob/master/sys/linux/socket_netlink_netfilter_queue.txt#L10-L14
If we do one template type for netlink msg with CMD and POLICY as arguments, then we can use it for all sendmsg calls in this file.
Please do such refactoring.